### ISSUE TITLE
deposit-ui: 400 (BAD REQUEST) when uploading a JSON file

### ISSUE THREAD

**Issue by krzysztof** [zenodo/zenodo#1453]
1. Create a JSON file with content:
```
{"test": "test"}
```
2. Upload will fail with "Error" (400 in chrome developer console)

**Comment by lnielsen** [comment]
Is there more details - we have some ~400 JSON files - see https://zenodo.org/search?page=1&size=20&q=filetype:json

**Comment by mittagessen** [comment]
I'm running into this issue right now. This doesn't seem to be caused by json files per se but uploading anything with a `.json/.JSON` extension causes an error on the UI. Let's take [this](https://zenodo.org/record/580101) deposition. Trying to create a new one with the 34.json in it on the sandbox instance doesn't work.

I can confirm that it doesn't happen on the (sandbox) API server but I haven't found an obvious source for the filtering in the UI code in `zenodo/modules/deposit/`.

**Comment by p-a-s-c-a-l** [comment]
I can confirm the problem.

Trying to a upload an arbitrary file with .json extension fails in sandbox even if I choose a JSON file previously downloaded from Zenodo (e.g. [oneVoxel.json](https://zenodo.org/record/2558396)):

![grafik](https://user-images.githubusercontent.com/1788226/52478086-86efbc80-2ba4-11e9-8663-309d2b37c067.png)

Omitting the .json extension in the filename (e.g [package_search[1]](https://ckan.myclimateservice.eu/api/3/action/package_search?fq=name%3Apublic-military-and-industrial-units)) does however work. Files uploaded are valid JSON documents.

**Comment by KelSolaar** [comment]
I came across this issue in the Sandbox.

**Comment by Herrner** [comment]
same here, but on production zenodo.org

**Comment by slint** [comment]
This is addressed by a fix in https://github.com/inveniosoftware/invenio-files-js/commit/20c11904fdbde46365634fa464ef07e698d7f2ea. Basically the issue is that the Content-Type of the upload request is not `application/octet-stream` which the `PUT /api/files/...` endpoint expects.

**Comment by KelSolaar** [comment]
Any word on when this will land on the server?

**Comment by equaeghe** [comment]
I'm hitting this and I really want the file extension to be `json`. Is there a *workaround* that allows for this now? (There is some urgency for me…) I can use the Firefox developer console if need be.

**Comment by emdupre** [comment]
Just wanted to ping my +1 on encountering this issue and see if there was an anticipated timeline for a patch -- or any current suggested work-around ! Thanks for all of your work, here.

**Comment by sveinugu** [comment]
Another ping!

I just did a thorough research of research data portals to use for depositing JSON files and concluded that Zenodo seemed to be the one with most features matching our requirements, only to experience that this issue that has gone unresolved for two years, even if it has been fixed upstream (if I understand correctly).

I was able to upload JSON files using the old files API (strangely without the header fix mentioned here: https://gitter.im/zenodo/zenodo?at=5df9ecda84c216770921c037), but the preview of the JSON files are still now working, which is our case is unfortunate: https://zenodo.org/record/3984947.

Any updates or possibilities for fixing soon?
